django/db/models/query.py

```diff
@@ -1306,13 +1307,14 @@ def _batched_insert(self, objs, fields, batch_size, ignore_conflicts=False):
         """
         Helper method for bulk_create() to insert objs one batch at a time.
         """
-        if ignore_conflicts and not connections[self.db].features.supports_ignore_conflicts:
+        connection = connections[self.db]
+        if ignore_conflicts and not connection.features.supports_ignore_conflicts:
             raise NotSupportedError('This database backend does not support ignoring conflicts.')
-        ops = connections[self.db].ops
+        ops = connection.ops
         max_batch_size = max(ops.bulk_batch_size(fields, objs), 1)
         batch_size = min(batch_size, max_batch_size) if batch_size else max_batch_size
         inserted_rows = []
-        bulk_return = connections[self.db].features.can_return_rows_from_bulk_insert
+        bulk_return = connection.features.can_return_rows_from_bulk_insert
```

Comment:
This one I'm iffy on. it's accessed 3 times, but it's also at the end of a conditional. Hoisting it up forces grabbing a reference to the connection in _all_ scenarios, rather than only checking it when `ignore_conflicts` is True. I think it'll be fine because only Oracle has `supports_ignore_conflicts = False` though?

Reply:
Yeah, it's fine.